controllers/gslb_controller.go

```diff
@@ -195,13 +198,82 @@ func (r *GslbReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			return nil
 		})
 
+	createGslbFromIngress := func(annotationKey string, annotationValue string, a handler.MapObject, strategy string) {
+		log.Info(fmt.Sprintf("Detected strategy annotation(%s:%s) on Ingress(%s)",
+			annotationKey, annotationValue, a.Meta.GetName()))
+		c := mgr.GetClient()
+		ingressToReuse := &v1beta1.Ingress{}
+		err := c.Get(context.Background(), client.ObjectKey{
+			Namespace: a.Meta.GetNamespace(),
+			Name:      a.Meta.GetName(),
+		}, ingressToReuse)
+		if err != nil {
+			log.Info(fmt.Sprintf("Ingress(%s) does not exist anymore. Skipping...", a.Meta.GetName()))
+			return
+		}
+		gslbExist := &k8gbv1beta1.Gslb{}
+		err = c.Get(context.Background(), client.ObjectKey{
+			Namespace: a.Meta.GetNamespace(),
+			Name:      a.Meta.GetName(),
+		}, gslbExist)
+		if err == nil {
+			log.Info(fmt.Sprintf("Gslb(%s) already exists. Skipping...", gslbExist.Name))
+			return
+		}
+		gslb := &k8gbv1beta1.Gslb{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:   a.Meta.GetNamespace(),
+				Name:        a.Meta.GetName(),
+				Annotations: a.Meta.GetAnnotations(),
+			},
+			Spec: k8gbv1beta1.GslbSpec{
+				Ingress: ingressToReuse.Spec,
+				Strategy: k8gbv1beta1.Strategy{
+					Type: strategy,
+				},
+			},
+		}
+
+		if strategy == "failover" {
+			for annotationKey, annotationValue := range a.Meta.GetAnnotations() {
+				if annotationKey == "k8gb.io/primarygeotag" {
+					gslb.Spec.Strategy.PrimaryGeoTag = annotationValue
+				}
+			}
+			if gslb.Spec.Strategy.PrimaryGeoTag == "" {
+				log.Info("k8gb.io/primarygeotag annotation is missing, skipping Gslb creation...")
+				return
+			}
+		}
+
+		log.Info(fmt.Sprintf("Creating new Gslb(%s) out of Ingress annotation", gslb.Name))
+		_ = c.Create(context.Background(), gslb)
+	}
+	ingressMapFn := handler.ToRequestsFunc(
+		func(a handler.MapObject) []reconcile.Request {
+			for annotationKey, annotationValue := range a.Meta.GetAnnotations() {
+				if annotationKey == "k8gb.io/strategy" {
+					switch annotationValue {
+					case "roundRobin":
+						createGslbFromIngress(annotationKey, annotationKey, a, "roundRobin")
+					case "failover":
```

Comment:
consider `"failover","roundRobin"` as constants

Reply:
@kuritka I thought about it but given I can't use constants as a case value(`case roundRobinConstant` will not work afaik) the constants introduction here looks a bit artificial

Reply:
@kuritka yeah, actually they work. Implemented constants in https://github.com/AbsaOSS/k8gb/pull/205/commits/86260ff365145f5e98c3a457280e0a9fadec0477